Sources/MapboxNavigation/CarPlayManager.swift

```diff
@@ -783,15 +752,33 @@ extension CarPlayManager: CPMapTemplateDelegate {
             return
         }
         
+        // Whenever panning interface is shown (either in preview mode or during active navigation),
+        // user should be given an opportunity to update buttons in `CPMapTemplate`.
         if let mapButtons = delegate?.carPlayManager(self,
                                                      mapButtonsCompatibleWith: traitCollection,
                                                      in: mapTemplate,
                                                      for: currentActivity) {
             mapTemplate.mapButtons = mapButtons
         } else {
-            let closeButton = carPlayMapViewController.panningInterfaceDismissalButton(for: mapTemplate)
-            carPlayMapViewController.dismissPanningButton = closeButton
-            mapTemplate.mapButtons = [closeButton]
+            if let carPlayMapViewController = carPlayMapViewController {
+                let closeButton = carPlayMapViewController.panningInterfaceDismissalButton(for: mapTemplate)
+                carPlayMapViewController.dismissPanningButton = closeButton
+                mapTemplate.mapButtons = [closeButton]
+            }
+        }
+        
+        if let leadingButtons = delegate?.carPlayManager(self,
```

Comment:
Whenever panning interface is shown we're only checking whether delegate methods provide any changes for `CPMapTemplate` buttons. If so - we use them, if not, originally set buttons will be used.